Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -634,27 +603,22 @@ private func resolveModuleAliases(packageBuilders: [ResolvedPackageBuilder],
         }
     }
 
-    guard hasAliases else { return nil }
+    guard hasAliases else { return }
     let aliasTracker = ModuleAliasTracker()
     for packageBuilder in packageBuilders {
-        for target in packageBuilder.package.targets {
-            for dep in target.dependencies {
-                if case let .product(prodRef, _) = dep,
-                   let prodPkg = prodRef.package {
-                    let prodPkgID = PackageIdentity.plain(prodPkg)
-                    // Track package ID dependency chain
-                    aliasTracker.addPackageIDChain(parent: packageBuilder.package.identity, child: prodPkgID)
-                    
-                    if let aliasList = prodRef.moduleAliases {
-                        for (depName, depAlias) in aliasList {
-                            // Track aliases for this product
-                            try aliasTracker.addAlias(depAlias,
-                                                      target: depName,
-                                                      product: prodRef.name,
-                                                      originPackage: PackageIdentity.plain(prodPkg),
-                                                      consumingPackage: packageBuilder.package.identity)
-                        }
-                    }
+        let targetDependencies = packageBuilder.package.targets.map {$0.dependencies}.flatMap{$0}
+        for dep in targetDependencies {
+            if case let .product(productRef, _) = dep,
+               let productPkg = productRef.package {
+                let productPkgID = PackageIdentity.plain(productPkg)
+                // Track dependency package ID chain
+                aliasTracker.addPackageIDChain(parent: packageBuilder.package.identity, child: productPkgID)
```

Comment:
not sure if it makes a difference, but should this be inside the if condition in 616?

Reply:
no this is intentional; need to keep track of the entire chain as well as its target list -- sometimes module aliases are not defined in a middle upstream package, but still need to be propagated up if there are targets in upstream that match the alias defined downstream